quic/s2n-quic-core/src/varint/tests.rs

```diff
@@ -4,24 +4,31 @@
 use super::*;
 use crate::varint::VarInt;
 use bolero::check;
-use s2n_codec::{assert_codec_round_trip_bytes, assert_codec_round_trip_value};
+use core::mem::size_of;
+use s2n_codec::{
+    assert_codec_round_trip_bytes, assert_codec_round_trip_value, DecoderBuffer, EncoderBuffer,
+};
 
 #[test]
 #[cfg_attr(miri, ignore)] // This test is too expensive for miri to complete in a reasonable amount of time
-fn round_trip() {
+fn round_trip_bytes_test() {
```

Comment:
Is Kani unable to handle this one?

Reply:
I tried it but it kept wanting a higher unwind value. I put `kani::unwind(50)` and it still complained. I also put a `kani::assume(bytes.len() <= 8)` and it didn't make a difference, unfortunately. I didn't want to block the rest of these proofs on it, though, but definitely worth exploring.

Reply:
I see. I'll investigate.

Reply:
It seems there's a loop that requires an unwinding greater than 256. Setting the unwind to 256 results in an unwinding assertion failure:
```
Check 941: std::array::try_collect_into_array::<std::iter::Map<&mut std::iter::Map<std::array::IntoIter<(), 256>, [closure@<[u8; 256] as kani::Arbitrary>::any::{closure#0}]>, fn(u8) -> std::ops::try_trait::NeverShortCircuit<u8> {std::ops::try_trait::NeverShortCircuit::<u8>}>, u8, std::ops::try_trait::NeverShortCircuitResidual, 256>.unwind.0
         - Status: FAILURE
         - Description: "unwinding assertion loop 0"
         - Location: ../../../../.rustup/toolchains/nightly-2022-11-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/array/mod.rs:884:21 in function std::array::try_collect_into_array::<std::iter::Map<&mut std::iter::Map<std::array::IntoIter<(), 256>, [closure@<[u8; 256] as kani::Arbitrary>::any::{closure#0}]>, fn(u8) -> std::ops::try_trait::NeverShortCircuit<u8> {std::ops::try_trait::NeverShortCircuit::<u8>}>, u8, std::ops::try_trait::NeverShortCircuitResidual, 256>
```
and with 257, it keeps unwinding till it runs out of memory :(

Reply:
Yeah that must be this
https://github.com/camshaft/bolero/blob/88834fbf32c371da710fb202b4f4bcce178b29fb/bolero-kani/src/lib.rs#L77